core/io/resource_loader.cpp

```diff
@@ -616,7 +616,7 @@ RES ResourceLoader::load(const String &p_path, const String &p_type_hint, bool p
 #ifdef TOOLS_ENABLED
 
 		res->set_edited(false);
-		if (timestamp_on_load) {
+		if (timestamp_on_load && !ProjectSettings::get_singleton()->is_using_datapack()) {
```

Comment:
I'm not sure this is the correct fix, as this would mean that if you load any data pack (like in your case an external one), modified time would no longer be updated anymore for files which are not in the data pack, since `ProjectSettings::using_datapack` seems to be all or nothing (it was meant for when the game is run from a `.pck`, but not designed to support running the game from source while using additional PCKs for more content).

The problem seems to be that `FileAccess` doesn't use `FileAccessPack` automatically here when trying to load a resource from the PCK (which would return 0 as modified time). I'm not sure how to solve that.

Reply:
Looking at `ProjectSettings::_load_resource_pack`, it seems ill-prepared for a scenario where you load PCK files from a non-PCK-based project in general, as it switches all future DirAccess to DirAccessPack (but apparently not for FileAccess).